index.bs

```diff
@@ -229,85 +303,176 @@ The {{XRHitTestResult/getPose(baseSpace)}} method, when invoked on {{XRHitTestRe
 
-Every time the user agent processes [=/XR animation frame=] for {{XRSession}}, it needs to <dfn>compute hit test results</dfn>. This MUST happen prior to invoking any of the callbacks from [=/list of animation frame callbacks=] of a given {{XRSession}}.
+Every time the user agent processes [=/XR animation frame=] for {{XRSession}}, it needs to <dfn>compute all hit test results</dfn>. This MUST happen prior to invoking any of the callbacks from [=/list of animation frame callbacks=] of a given {{XRSession}}.
+
+<div class="algorithm" data-algorithm="compute-all-hit-test-results">
+
+In order to [=compute all hit test results=] for a given {{XRFrame}} |frame|, the user agent MUST perform the following steps:
+    1. Invoke [=compute hit test results=] algorithm with |frame|.
+    1. Invoke [=compute hit test results for transient input=] algorithm with |frame|.
+
+</div>
 
 <div class="algorithm" data-algorithm="compute-hit-test-results">
 
-In order to [=compute hit test results=] for a given {{XRFrame}} |frame|, for each hit test source, |hitTestSource|, that is present in {{XRFrame/session}}'s [=XRSession/set of active hit test sources=], the user agent MUST perform the following steps:
+In order to <dfn>compute hit test results</dfn> for a given {{XRFrame}} |frame|, for each hit test source, |hitTestSource|, that is present in {{XRFrame/session}}'s [=XRSession/set of active hit test sources=], the user agent MUST perform the following steps:
     1. Let |entityTypes| be the |hitTestSource|'s [=XRHitTestSource/entity types=].
     1. Let |session| be |frame|'s {{XRFrame/session}}.
     1. Let |device| be the |session|'s  [=XRSession/XR device=].
     1. Query the |device|'s tracking system for |hitTestSource|'s [=XRHitTestSource/native origin=]'s latest |coordinates|.
     1. Interpret |hitTestSource|'s [=XRHitTestSource/offset ray=], |offsetRay|, as if expressed relative to |coordinates| and using that interpretation, perform [=native hit test=] obtaining [=native hit test results=] |nativeResults|.
     1. Let |hitTestResults| be an empty [=/list=].
-    1. Given the |frame| and |entityTypes|, for each native hit test result |nativeResult| in |nativeResults|, perform the following steps:
-        1. [=Create a hit test result=], |hitTestResult|.
+    1. For each native hit test result |nativeResult| in |nativeResults|, perform the following steps:
+        1. [=Create a hit test result=], |hitTestResult| from |frame|, |entityTypes|, and |nativeResult|.
         1. If |hitTestResult| is <code>null</code>, continue to the next entry in |nativeResults|.
         1. Add |hitTestResult| to |hitTestResults| such that the list remains sorted by the [=distance along the ray=] from |offsetRay| to |nativeResult|.
     1. Store |hitTestResults| in |frame|'s [=XRFrame/map of hit test sources to hit test results=] under the |hitTestSource| key.
 
 </div>
 
+<div class="algorithm" data-algorithm="compute-hit-test-results-for-transient-input">
```

Comment:
Related to the above point of some transient input sources only existing for the space of a select event, I want to clarify the intended "lifetime" of a hit test result. This algorithm suggests that hit test results can only be returned while the input source is in the active list, right? The bit about animation frames above made me wonder if the intent was to return all hit test results that occurred since the previous animation frame, regardless of if the input source was still available.

Reply:
> This algorithm suggests that hit test results can only be returned while the input source is in the active list, right?

Correct. Chrome's case is a bit specific because as far as I can tell, our transient input sources always are alive for at least one animation frame (so results are available both for the select events, and during rAF callbacks), but the intent here is for the API to not hand out objects that are already in zombie state (transient hit test result contains XRInputSource so I don't think we want to deliver results whose the input source is no longer active).

Reply:
To clarify, the transient input sources in AR case are alive for at least one animation frame.